client/interface.go

```diff
@@ -38,6 +38,7 @@ type Result interface {
 	Round() uint64
 	Randomness() []byte
 	Signature() []byte
+	SignatureV2() []byte
```

Comment:
exposing this on the interface is gonna be awkward:
* once we eventually deprecate `Signature`, we'd really like `SignatureV2` to become signature, rather than having that be a permanent interface name.

Proposal instead:
This interface is the same, to configure it, we provide an additional configuration Option that can be passed at client construction that is: `WithV2SignaturesAfter(round)`, and Signature when used on a result above that defined round will use the v2 signature.

Reply:
Okay. I stared at this a bit more

* Agree that we do need to expose both v1 and v2 signatures so that we can pass them through relays / etc. Having both fields on `RandomData` is good.

* The way we're exposing `Randomness` is the ideal, where it is computed off the v2 signature above the configured round for the client. We should ideally have a single `Signature` interface that does the same.

The motivation for that, is that downstream consumers ([example](https://github.com/filecoin-project/lotus/blob/master/chain/beacon/drand/drand.go#L149) ) use `.Signature()` and we'd like that to switch in the same way.

I think the root pain is that we've coupled the `Result` interface as what methods are exposed to consumers with our serialization of rounds for passing them through the system.